From 4303525f58e56e4ea9cb7c9c5d8944f5122705fd Mon Sep 17 00:00:00 2001
From: Glenn Morris <rgm@gnu.org>
Date: Sun, 9 Nov 2014 23:12:37 -0800
Subject: Nil load-path elements shouldn't crash Emacs

This upstream patches has been added:

  * lisp/startup.el (command-line): Handle nil elements in load-path.

Origin: upstream, http://git.savannah.gnu.org/cgit/emacs.git/commit/?id=edfdb22f674312389ccf5d5e37efa4d3f1516994
Bug-Debian: http://bugs/debian.org/768751
Added-by: Rob Browning <rlb@defaultvalue.org>
---
 lisp/ChangeLog  | 4 ++++
 lisp/startup.el | 1 +
 2 files changed, 5 insertions(+)

diff --git a/lisp/ChangeLog b/lisp/ChangeLog
index 4c9b60e..334c73f 100644
--- a/lisp/ChangeLog
+++ b/lisp/ChangeLog
@@ -1,3 +1,7 @@
+2014-11-10  Glenn Morris  <rgm@gnu.org>
+
+	* startup.el (command-line): Handle nil elements in load-path.
+
 2014-10-20  Glenn Morris  <rgm@gnu.org>
 
 	* Version 24.4 released.
diff --git a/lisp/startup.el b/lisp/startup.el
index c4f9f79..1ef2556 100644
--- a/lisp/startup.el
+++ b/lisp/startup.el
@@ -1336,6 +1336,7 @@ the `--debug-init' option to view a complete error backtrace."
   (let (warned)
     (dolist (dir load-path)
       (and (not warned)
+	   (stringp dir)
 	   (string-match-p "/[._]emacs\\.d/?\\'" dir)
 	   (string-equal (file-name-as-directory (expand-file-name dir))
 			 (expand-file-name user-emacs-directory))
